File: LeetCode Solutions/Problems/2341. Maximum Number of Pairs in Array.py
class Solution:
    def numberOfPairs(self, nums: List[int]) -> List[int]:
        c = Counter(nums)
        pairs = 0
        remaining = 0
        for key, value in c.items():
            if value % 2 == 0:
                count = value // 2
                pairs += count
            else:
                count = (value + 1) // 2
                pairs += count - 1
                remaining += 1
        return [pairs, remaining]

File: LeetCode Solutions/Problems/2341. Maximum Number of Pairs in Array.java
class Solution {
    public int[] numberOfPairs(int[] nums) {
        int []result = new int [2];
        int [] temp = new int[101];
        for (int i:nums){
            temp[i]++;
        }
        int count = 0;
        for (int i=0; i<101; i++){
            if (temp[i] % 2 != 0){
                count++;
            }
        }
        result[0] = (nums.length-count)/2;
        result[1] = count;
       return result;
    }
}